api/xds_protocol.rst

```diff
@@ -469,3 +469,158 @@ occurred via a :ref:`resource update <xds_protocol_resource_update>`.
+configuration. The management server will respond with a
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+with the :ref:`resources <envoy_api_field_DeltaDiscoveryResponse.resources>`
+field populated with the specified virtual hosts, the 
+:ref:`name <envoy_api_field_resource.name>` field populated with
+the virtual host name, and the 
+:ref:`aliases <envoy_api_field_resource.aliases>` field
+populated with the explicit (no wildcard) domains of the virtual host. Future
+updates to these virtual hosts will be sent via spontaneous updates.
+
+Updates to the route configuration entry to which a virtual host belongs will
+clear the virtual host table and require all virtual hosts to be sent again. It
+may be useful for the management server to populate and RDS responses with the
+subscribed list of virtual hosts. 
+
+Joining/Reconnecting to a management server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When an Envoy instance forms a delta xDS connection to the xDS management
+server for the first time, it will receive a base configuration filtered down
+to a subset of routes that are likely useful to the Envoy instance. For
+example, these might be filtered down by routes that exist in the current
+namespace. The xDS management server will send this same collection of
+resources on reconnect.
+
+.. figure:: diagrams/delta_rds_connection.svg
+   :alt: Delta RDS connection
+
+Virtual Host Naming Convention
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Virtual hosts in VHDS are identified by a combination of the name of the route
+configuration to which the virtual host belong as well as the host HTTP "host"
+header (authority for HTTP2) entry. Resources should be named as follows:
+
+<route configuration name>/<host entry>
+
+Note that matching should be done from right to left since a host entry cannot
+contain slashes while a route configuration name can. 
+
+Requesting Additional resources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>` with the
+:ref:`resource_names_subscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_subscribe>`
+field populated with the route config names
++ domains of each of the resources that it would like to subscribe to. Each of
+the virtual hosts contained in the
+:ref:`DeltaDiscoveryRequest's <envoy_api_msg_DeltaDiscoveryRequest>`
+resources field will be added to the route configuration maintained by Envoy.
+If Envoy's route configuration already contains a given virtual host, it will
+be overwritten by data received in the
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+but only if the updated virtual host is different from its current state.
+The :ref:`resource.aliases <envoy_api_field_resource.aliases>` field contains
+all host/authority header values used to create the on-demand request. During
+spontaneous updates configuration server will only send updates for virtual
```

Comment:
I think we do allow the management server to send more than the known hosts; it's not generally useful but it's allowed.